packages/react/src/components/header/components/navigationLink/NavigationLink.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { cloneElement, useCallback, useContext, useEffect, useRef, useState } from 'react';
+
+// import core base styles
+import 'hds-core';
+import styles from './NavigationLink.module.scss';
+import classNames from '../../../../utils/classNames';
+import { Link } from '../../../link';
+import { NavigationLinkDropdown, NavigationLinkInteraction, DropdownMenuPosition } from './navigationLinkDropdown';
+import { HeaderNavigationMenuContext } from '../headerNavigationMenu/HeaderNavigationMenuContext';
+import { DropdownDirection } from './types';
+
+export type NavigationLinkProps = Omit<
+  React.ComponentPropsWithoutRef<'a'>,
+  'target' | 'href' | 'onPointerEnterCapture' | 'onPointerLeaveCapture' | 'aria-label'
+> & {
+  /**
+   * Indicator for active link. This is used in HeaderNavigationMenu.
+   */
+  active?: boolean;
+  /**
+   * Additional class names to apply.
+   */
+  className?: string;
+  /**
+   * Set the direction where the dropdown should appear. Use DropdownDirection.Dynamic for nested dropdowns as it sets the dropdown menu to the right but if there's no space it'll put it to the left.
+   * @default DropdownDirection.Down;
+   */
+  dropdownDirection?: DropdownDirection;
+  /**
+   * Array of NavigationLink components to render in a dropdown. Can be used only inside navigation components.
+   */
+  dropdownLinks?: Array<React.ReactElement>;
+  /**
+   * Hypertext Reference of the link.
+   */
+  href: string;
+  /**
+   * Element index given by parent mapping.
+   * @internal
+   */
+  index?: number;
+  /**
+   * Label for link.
+   */
+  label: string;
+  /**
+   * Which sub navigation index is open.
+   * @internal
+   */
+  openSubNavIndex?: number;
+  /**
+   * Set which sub navigation index is open.
+   * @internal
+   */
+  setOpenSubNavIndex?: (val: number) => void;
+  /**
+   * Size of the link.
+   */
+  size?: 'S' | 'M' | 'L';
```

Comment:
I think we should start normalizing these size string props' required type formats at some point. There are currently at least three different variations of the size prop's format:

- lowercase s/m/l
- uppercase S/M/L
- sm/md/lg

It might get confusing for people using these components at some point even if these are documented well.

Reply:
Very good idea, definitely needs to be done for consistency's sake.

Reply:
Welp, I happened to find only smL/SML variants and only Link uses upper case so we have to use it here, exceeeept it's not used at the moment so we can leave it out! 💡 It's weird that Link uses S/M/L because then it maps it to smaller case because it uses the same prop to icons 😆